roles/confluent.variables/defaults/main.yml

```
@@ -307,13 +307,27 @@ kafka_broker_custom_listeners:
     ssl_mutual_auth_enabled: "{{ ssl_mutual_auth_enabled }}"
     sasl_protocol: "{{ sasl_protocol }}"
 
+kafka_broker_control_plane_listener:
+  controller:
+    name: CONTROLLER
+    port: 9089
+    ssl_enabled: "{{ ssl_enabled }}"
+    ssl_mutual_auth_enabled: "{{ ssl_mutual_auth_enabled }}"
+    sasl_protocol: "{{ sasl_protocol }}"
+
 # Deprecated variable
 kafka_broker_configure_additional_brokers: true
 
 ### Boolean to configure more than one kafka listener. Defaults to true. NOTE- kafka_broker_configure_additional_brokers is deprecated
 kafka_broker_configure_multiple_listeners: "{{kafka_broker_configure_additional_brokers}}"
 
-kafka_broker_listeners: "{{ kafka_broker_default_listeners | combine(kafka_broker_custom_listeners, recursive=True) if kafka_broker_configure_multiple_listeners|bool else  kafka_broker_default_listeners}}"
+### Boolean to configure control plane listener on separate port. Applied only if kafka_broker_configure_multiple_listeners is true
+kafka_broker_configure_control_plane_listener: false
+
+### Control Planer listener name.
+kafka_broker_control_plane_listener_name: "controller"
+
+kafka_broker_listeners: "{{ kafka_broker_default_listeners | combine(kafka_broker_custom_listeners | combine(kafka_broker_control_plane_listener,recursive=True) if kafka_broker_configure_control_plane_listener|bool else kafka_broker_custom_listeners, recursive=True) if kafka_broker_configure_multiple_listeners|bool else  kafka_broker_default_listeners}}"
```

I really wish there was a cleaner way to do this... at the minimum can you split it over a few lines so its easier to read. I'm going to try out something as well

```suggestion
kafka_broker_listeners: "{{ kafka_broker_default_listeners | combine(kafka_broker_custom_listeners | combine(kafka_broker_control_plane_listener,recursive=True) if kafka_broker_configure_control_plane_listener|bool 
else kafka_broker_custom_listeners, recursive=True) if kafka_broker_configure_multiple_listeners|bool 
else  kafka_broker_default_listeners}}"
```

---

Yes I wish it too. I didn't know variables can go on multiple lines when in curly braces. Changed as per suggestion.